### PR TITLE
fix invalid basic auth secret behaviour

### PR DESCRIPTION
**Pull request by dianfishekqi**
## what

Added a conditional to not create the basic auth secret if `basicAuthSecretName` is given, added the field to values

## why

Because before this change, even if u set basicAuthSecretName u have to set basicAuth.name and basicAuth.password for basic auth to work.

### CODE REVIEW

**Comment by GenPage** [comment]
@dianfishekqi Did you do any testing by generating the templates locally?

**Comment by GenPage** [comment]
@dianfishekqi Thank you for confirming, can you rebase and we'll wait for some chart index fixes in #253 before merging

**Comment by dianfishekqi** [comment]
@GenPage rebased

**Comment by GenPage** [comment]
@dianfishekqi you'll need to bump the chart version again

**Comment by dianfishekqi** [comment]
@GenPage done

File: charts/atlantis/values.yaml
Comment (nitrocode) [inline charts/atlantis/values.yaml:177]: Please update the readme

**Comment by GenPage** [comment]
@dianfishekqi Thank you, we're just waiting on #253 to merge first before we merge a new release.

**Comment by nxf5025** [comment]
Is this good to merge & release now that #253 is released?

**Comment by dianfishekqi** [comment]
@GenPage  ?

**Review by GenPage — APPROVED** [review]

**Comment by GenPage** [comment]
@nxf5025 @dianfishekqi There was other investigating in the chart-releaser I was doing other than #253. See #255.

**Pull request merged**
